lib/Jsrt/Jsrt.cpp

```diff
@@ -69,12 +80,28 @@ JsErrorCode CheckContext(JsrtContext *currentContext, bool verifyRuntimeState, b
     return JsNoError;
 }
 
-CHAKRA_API JsCreateRuntime(_In_ JsRuntimeAttributes attributes, _In_opt_ JsThreadServiceCallback threadService, _Out_ JsRuntimeHandle *runtimeHandle)
+template <class Fn>
+bool CallbackWrapper(Fn fn)
 {
-    return GlobalAPIWrapper([&] () -> JsErrorCode {
+    return CallbackWrapper(fn, false);
+}
+
+/////////////////////
+
+//A create runtime function that we can funnel to for regular and record or debug aware creation
+JsErrorCode CreateRuntimeCore(_In_ JsRuntimeAttributes attributes, _In_opt_ wchar_t* optRecordUri, _In_opt_ wchar_t* optDebugUri, _In_ UINT32 snapInterval, _In_ UINT32 snapHistoryLength, _In_opt_ JsThreadServiceCallback threadService, _Out_ JsRuntimeHandle *runtimeHandle)
+{
+    return GlobalAPIWrapper([&]() -> JsErrorCode {
         PARAM_NOT_NULL(runtimeHandle);
         *runtimeHandle = nullptr;
 
+#if ENABLE_TTD
+        if(optRecordUri != nullptr || optDebugUri != nullptr)
+        {
+            attributes = static_cast<JsRuntimeAttributes>(attributes | JsRuntimeAttributeEnableExperimentalFeatures);
+        }
```

Comment:
Good call, I had this to ensure consistency with Node. Moved to Ch since that is where is should be.